java-checks/src/main/java/org/sonar/java/checks/unused/UnusedMethodParameterCheck.java

```diff
@@ -95,23 +104,53 @@ public void visitNode(Tree tree) {
       .filter(id -> !unresolvedIdentifierNames.contains(id.name()))
       .collect(Collectors.toList());
     if (!unused.isEmpty()) {
-      reportUnusedParameters(unused);
+      reportUnusedParameters(methodTree, unused);
     }
   }
 
-  private void reportUnusedParameters(List<IdentifierTree> unused) {
-    List<JavaFileScannerContext.Location> locations = new ArrayList<>();
-    for (IdentifierTree identifier : unused) {
-      locations.add(new JavaFileScannerContext.Location("Remove this unused method parameter " + identifier.name() + "\".", identifier));
-    }
+  private void reportUnusedParameters(MethodTree methodTree, List<IdentifierTree> unused) {
     IdentifierTree firstUnused = unused.get(0);
-    String msg;
-    if (unused.size() > 1) {
-      msg = "Remove these unused method parameters.";
-    } else {
-      msg = "Remove this unused method parameter \"" + firstUnused.name() + "\".";
+    List<JavaFileScannerContext.Location> secondaryLocations = unused.stream()
+      .skip(1)
+      .map(identifier -> new JavaFileScannerContext.Location(String.format(SECONDARY_MESSAGE_FORMAT, identifier.name()), identifier))
+      .collect(Collectors.toList());
+    String parameterNames = unused.stream().map(identifier -> "\"" + identifier.name() + "\"").collect(Collectors.joining(", "));
+    ((InternalJavaIssueBuilder) ((DefaultJavaFileScannerContext) context).newIssue())
+      .forRule(this)
+      .onTree(firstUnused)
+      .withMessage(unused.size() > 1 ?  PRIMARY_PLURAL_MESSAGE_FORMAT : PRIMARY_SINGULAR_MESSAGE_FORMAT, parameterNames)
+      .withSecondaries(secondaryLocations)
+      .withQuickFixes(() -> createQuickFixes(methodTree, unused))
+      .report();
+  }
+
+  private List<JavaQuickFix> createQuickFixes(MethodTree methodTree, List<IdentifierTree> unused) {
+    List<JavaQuickFix> quickFixes = new ArrayList<>();
+    List<VariableTree> parameters = methodTree.parameters();
+    for (int i = 0; i < parameters.size(); i++) {
+      VariableTree parameter = parameters.get(i);
+      if (unused.contains(parameter.simpleName())) {
+        boolean isFirst = i == 0;
+        boolean isLast = i == parameters.size() - 1;
+        AnalyzerMessage.TextSpan textSpanToRemove;
+        if (isFirst && isLast) {
+          // no need to remove any comma
+          textSpanToRemove = AnalyzerMessage.textSpanBetween(methodTree.openParenToken(), false, methodTree.closeParenToken(), false);
+        } else if (isLast) {
+          // also remove the previous comma and spaces between previousParameter and parameter
+          VariableTree previousParameter = parameters.get(i - 1);
+          textSpanToRemove = AnalyzerMessage.textSpanBetween(previousParameter.endToken(), true, methodTree.closeParenToken(), false);
+        } else {
+          // also remove the next comma and spaces between parameter and nextParameter
```

Comment:
The approach is sound here provided that all quickfixes are not applied in at once. Otherwise. we need to think of how to  generate the list of contiguous segments to delete.